src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1526,6 +1526,23 @@ mongoc_uri_finalize_directconnection (mongoc_uri_t *uri, bson_error_t *error)
    return true;
 }
 
+static void
+mongoc_uri_check_for_genuine_hosts (const char *hosts)
+{
+   char *hosts_lowercase = lowercase_str_new (hosts);
+
+   if (strstr (hosts_lowercase, ".cosmos.azure.com")) {
+      MONGOC_INFO ("Azure Cosmos DB is not genuine");
```

Comment:
Note: these messages are not finalized. My understanding is that we're waiting on URLs to include here. I believe @alexbevi is our point of contact on that.

Reply:
Correct. We have WEBSITE-15470 tracking that for now.